### PR TITLE
extend innkeeper marshaling with custom predicates

### PR DESCRIPTION
**Pull request by aryszka**

### CODE REVIEW

File: innkeeper/client_test.go [hunk 1]
```diff
@@ -296,6 +296,15 @@ func TestParsingInnkeeperComplexRoute(t *testing.T) {
...
 					}]
 				},
+                "predicates": [{
+                    "name": "customPredicate1"
+                }, {
+                    "name": "customPredicate2",
+                    "args": []
+                }, {
+                    "name": "customPredicate3",
+                    "args": ["string1", 3.14]
```
Comment (danpersa) [inline innkeeper/client_test.go]: do we really need float numbers? for filters we only integers (at least in Innkeeper)

Comment (aryszka) [inline innkeeper/client_test.go]: why only integers?

Comment (aryszka) [inline innkeeper/client_test.go]: does innkeeper support 64 bit integers only?

Comment (aryszka) [inline innkeeper/client_test.go]: udpated to check for 64 bit integers. please, let us know if that's not ok.

Comment (aryszka) [inline innkeeper/client_test.go]: after consulting dan: limited the innkeeper numbers to 32 bit, signed integers

File: innkeeper/client_test.go [hunk 2]
```diff
@@ -479,6 +501,33 @@ func TestConvertRouteHeaders(t *testing.T) {
...
 
+func TestConvertRoutePredicates(t *testing.T) {
+	d := &routeData{Route: routeDef{Predicates: []customPredicate{{
+		Name: "customPredicate1",
+	}, {
+		Name: "customPredicate2",
+		Args: []interface{}{},
+	}, {
+		Name: "customPredicate3",
+		Args: []interface{}{"string1", 3.14},
```
Comment (danpersa) [inline innkeeper/client_test.go]: shouldn't this be an integer?

File: innkeeper/client_test.go (hunk 1 above)
Comment (danpersa) [inline innkeeper/client_test.go]: shouldn't line 306 have an integer there?

**Comment by danpersa** [comment]
lgtm

**Pull request merged**